lib/abr/simple_abr_manager.js

```diff
@@ -44,8 +44,14 @@ shaka.abr.SimpleAbrManager = class {
 
     /** @private {shaka.abr.EwmaBandwidthEstimator} */
     this.bandwidthEstimator_ = new shaka.abr.EwmaBandwidthEstimator();
-    // TODO: Consider using NetworkInformation's change event to throw out an
-    // old estimate based on changing network types, such as wifi => 3g.
+
+    // Some browsers implement the Network Information API, which allows
+    // retrieving information about a user's network connection.
+    if (navigator.connection) {
+      navigator.connection.addEventListener('change', () => {
+        this.bandwidthEstimator_ = new shaka.abr.EwmaBandwidthEstimator();
```

Comment:
This seems like a reasonable way to reset the estimation, but at the moment this occurs, we will use the default estimate.  That default was captured when the Player was created or its configuration reset, and with a change of network, we should probably get a new default estimate.

Therefore I would propose that you take some of the logic from `shaka.util.PlayerConfiguration.createDefault` and move it into `SimpleAbrManager`.  Instead of the default bandwidth estimate in the config being set based on NetworkInformation, it would be static in the config and overridden here by NetworkInfo.  We would re-read that value in the constructor and on network change.

This is what was meant by the TODO in `createDefault`:

```js
      // TODO: Move this into AbrManager, where changes to the estimate can be
      // observed and absorbed.
```

Since this would be overriding not only the default config, but any explicit config from the user, I think we should also offer a config flag to say whether or not the NetworkInfo API should be used for this (default true).

Reply:
I made the change!